apps/organizations/views.py

```diff
@@ -223,6 +224,52 @@ def handle_stripe_account_link(request, rp_pk):
     return Response(data)
 
 
+@create_revision()
+@api_view(["POST"])
+# TODO: add correct permissions
+@permission_classes([IsAuthenticated])
+def handle_send_test_email(request):
+    """This endpoint sends test emails to the user so that they can see what it looks like.
+    Available email types are:
+
+    1. receipt: this is the email that is sent to the user when they make a payment
+    2. reminder: this is the email that is sent to the user reminding of a future payment
+    3. magic_link: this is the email that is sent to the user when they click on the magic link to log in the contributor portal
+    """
+    rp_pk = request.data["revenue_program"]
+    email_type = request.data["email_type"]
+
+    revenue_program = get_object_or_404(RevenueProgram, pk=rp_pk)
+    if not request.user.is_superuser:
+        if not request.user.roleassignment.can_access_rp(revenue_program):
+            logger.warning(
+                (
+                    "[handle_send_test_email] was asked to send a test email link for RP with ID %s by user with id %s who does "
+                    "not have access."
+                ),
+                rp_pk,
+                request.user.id,
+            )
+            raise PermissionDenied(f"You do not have permission to access revenue program with the PK {rp_pk}")
+
+    if email_type not in ["receipt", "reminder", "magic_link"]:
+        raise ValidationError(f"Invalid email type {email_type}")
+
+    logger.info("Sending test email with type '%s', for user %s and rp %s", email_type, request.user.id, rp_pk)
+
+    if email_type == "receipt":
+        data = make_send_test_thank_you_email_data(request.user, revenue_program)
+        send_thank_you_email.delay(data)
+    elif email_type == "reminder":
+        # TODO: send reminder email
+        data = ""
+    elif email_type == "magic_link":
+        # TODO: send magic link email
+        data = ""
```

Comment:
Could use a `match` statement here.
```suggestion
    match email_type:
        case "receipt":
            data = make_send_test_thank_you_email_data(request.user, revenue_program)
            send_thank_you_email.delay(data)
            .... etc.
```